app/Jobs/ProcessSubmission.php

```diff
@@ -53,7 +53,9 @@ private function renameSubmissionFile($src, $dst): bool
         if (config('cdash.remote_workers')) {
             $url = config('app.url') . '/api/v1/deleteSubmissionFile.php';
             $client = new \GuzzleHttp\Client();
-            $response = $client->request('DELETE', $url, ['query' => ['filename' => $src, 'dest' => $dst]]);
+            $response = $client->request('DELETE', $url, [
+                'query'   => ['filename' => encrypt($src), 'dest' => encrypt($dst)],
+            ]);
```

Comment:
The `encrypt()` helper can be applied to entire arrays.  I think it would be cleaner to just encrypt the entire structure like this:

```suggestion
            $response = $client->request('DELETE', $url, 
                'query' => encrypt([
                    'filename' => $src,
                    'dest' => $dst
                ]),
            ]);
```

Apply throughout.

Reply:
Encrypting the whole query array results in requests like this:

```
DELETE /CDash/api/v1/deleteSubmissionFile.php?eyJpdiI6IjNUOHFNTWNsYllZRmgrbGZJZFBFZXc9PSIsInZhbHVlIjoicVpFTnlWRTU4UXVKSkZDNVErMjJqUC9SaUF4Uzlsd3JtNExKWWhhQ2s1NmhBZ3FzTnUyYWlyNUpIajJaVFRpVWRDWHRCSmlqWGxWajBCV0g4blBrNndqQURYUCt0N0tEZHZiVk05R2hwQTdiR045eCtXUmRMdlRMZlN4aUR3Z2ZlTnl4ZGtackxXWXlYd3V2RWErVHFyYjlPTGF2Sm1TTlRpQ3dOZzZ2RVF4dHFWK0cxTFVzRk9JOVhoSm5PdUF0WHNQMVhQbk9wV1VWUDR4OGsyZTNIZXJHeDlwRHB6ZG9VckpLSzZGUzJTUVowc3hXbEtvMERjbWpEMmErczJOdSsvYURISHAybE9sY3laM056dHNaalBES3hiY1FZdlNkZ0dnMkU0QW5pY2VHenJhVzlNMUl6R3B2ZXJQOGtaTEc4Tk1ES21iOE85RWZxc3NsUk5va3V5RTE2UFhPNzBiVnh0K2V2b0NXSjNBPSIsIm1hYyI6IjBmNTYzZDJiYzk3YjlhM2JhZDRiZWJiM2IwNjdiNGVkMDA2MjVhNzM1NThmZTZjYzM0YjhkYTVlMzc2M2Y2NDIiLCJ0YWciOiIifQ==
```

This is more complicated to decrypt from the API endpoint.

The following changes seems to work, but I think it's messier / more fragile than encrypting the individual parameters:

```php
$request_array = decrypt((string) array_key_first($_REQUEST));
```

Any other thoughts on how to achieve this?

Reply:
Good point.  I guess it's probably just easier to leave it as-is for now.  This kind of thing would be better handled by dedicated middleware.  I'll add that to my list of future improvements.